docs/static/running-logstash-command-line.asciidoc

```diff
@@ -101,10 +101,6 @@ With this command, Logstash concatenates three config files, `/tmp/one`, `/tmp/t
   If you wish to use both defaults, please use the empty string for the `-e` flag.
   The default is nil.
 
-*`--java-execution`*::
```

Comment:
Latergram:

I think we can do to make this transition easier for our users, especially for those who opt for explicit configuration or who control their configuration with environment variables.

 - `7.x`: the `false` value for `--java-execution` or `java.execution` is deprecated. We emit a deprecation log that REE is deprecated and will be removed in "a future major release of Logstash". To solve this deprecation warning, the user can either remove the flag or set its value to `true`.
 - `8.x`: the `--java-execution` or `java.execution` option _itself_ is deprecated, and only accepts the `true` value. When this is specified, we emit a deprecation log about the option being removed in "a future major release of Logstash"

The primary goal that this path accomplishes is that explicit configuration from Logstash `7.$last` that contains no deprecations works without modification in `8.0.0`. It allows our users to be explicit in their configuration of `7.x` and does not require an intermediate step where they are required to remove settings from their config that are meaningful in Logstash 7 prior to upgrading to Logstash 8.

The other transition option is to fast-track the removal of the setting, but I think that may be less clear to the user.

 - `7.x`: the `--java-execution` or `java.execution` option _itself_ is deprecated.
   - When this option is specified as `true`, we emit a deprecation log indicating that it is not needed and the option will be removed in "a future major release of Logstash".
   - When the option is specified as `false`, we emit a different deprecation log indicating that both REE and the option will be removed in "a future major release of Logstash".

Note that the fast-track will make environment variable control of the config file not possible.